docs/decisions/0024-connectors-api-equalization.md

```diff
@@ -0,0 +1,234 @@
+}
+
+public static class ChatCompletionExtensions
+{
+    //                       v Single          vv Standardized Prompt (Parse <message> tags)
+    public static async Task<ChatContent> GetChatContentAsync(string prompt, ...);
+
+    //                       v Single
+    public static async Task<ChatContent> GetChatContentAsync(ChatHistory chatHistory, ...);
+}
+```
+
+### ITextCompletion
+
+Before:
+
+```csharp
+public interface ITextCompletion : IAIService
+{
+    Task<IReadOnlyList<ITextResult>> GetCompletionsAsync(string prompt, ...);
+
+    IAsyncEnumerable<T> GetStreamingContentAsync<T>(string prompt, ...);
+}
+
+public static class TextCompletionExtensions
+{
+    public static async Task<string> CompleteAsync(string text, ...);
+
+    public static IAsyncEnumerable<StreamingContent> GetStreamingContentAsync(string input, ...);
+}
+```
+
+After:
+
+```csharp
+public interface ITextCompletion : IAIService
+{
+    Task<IReadOnlyList<TextContent>> GetTextContentsAsync(string prompt, ...);
+
+    IAsyncEnumerable<StreamingTextContent> GetStreamingTextContentsAsync(string prompt, ...);
+}
+
+public static class TextCompletionExtensions
+{
+    public static async Task<TextContent> GetTextContentAsync(string prompt, ...);
+}
+```
+
+## Content Abstractions
+
+### Model Comparisons
+
+#### Current Streaming Abstractions
+
+| Streaming (Current)                         | Specialized\* Streaming (Current)                               |
+| ------------------------------------------- | --------------------------------------------------------------- |
+| `StreamingChatContent` : `StreamingContent` | `OpenAIStreamingChatContent`                                    |
+| `StreamingTextContent` : `StreamingContent` | `OpenAIStreamingTextContent`, `HuggingFaceStreamingTextContent` |
+
+#### Non-Streaming Abstractions (Before and After)
+
+| Non-Streaming (Before)        | Non-Streaming (After)          | Specialized\* Non-Streaming (After)           |
+| ----------------------------- | ------------------------------ | --------------------------------------------- |
+| `IChatResult` : `IResultBase` | `ChatContent` : `ModelContent` | `OpenAIChatContent`                           |
+| `ITextResult` : `IResultBase` | `TextContent` : `ModelContent` | `OpenAITextContent`, `HuggingFaceTextContent` |
+| `ChatMessage`                 | `ChatContent` : `ModelContent` | `OpenAIChatContent`                           |
+
+_\*Specialized: Connector implementations that are specific to a single AI Service._
+
+### New Non-Streaming Abstractions:
+
+`ModelContent` was chosen to represent a `non-streaming content` top-most abstraction which can be specialized and contains all the information that the AI Service returned. (Metadata, Raw Content, etc.)
+
+```csharp
+/// <summary>
+/// Base class for all AI non-streaming results
+/// </summary>
+public abstract class ModelContent
```

Comment:
What's the logic behind this name? Aren't streaming results also coming back from a model?

Reply:
Agreement
```
StreamingContent -> StreamingContentBase
ModelContent -> ContentBase
```